core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -818,6 +858,37 @@ private static AnnotatedValueResolver ofInjectableTypes0(String name, AnnotatedE
         };
     }
 
+    /**
+     * Returns an attribute resolver which retrieves a value specified by {@code attrKeys}
+     * from the {@link RequestContext}.
+     */
+    private static BiFunction<AnnotatedValueResolver, ResolverContext, Object>
+    attributeResolver(Iterable<AttributeKey<?>> attrKeys) {
+        return (resolver, ctx) -> {
+            Class<?> targetType = resolver.rawType();
+            if (targetType.isPrimitive()) {
+                targetType = Primitives.wrap(targetType);
+            }
+
+            for (AttributeKey<?> attrKey : attrKeys) {
+                final Object value = ctx.context.attr(attrKey);
+                if (value != null) {
+                    final boolean isValidType = targetType.isInstance(value);
+
+                    if (!isValidType) {
```

Comment:
This is simple enough without a variable.
```suggestion
                    if (!targetType.isInstance(value)) {
```